### PR TITLE
CHM296-km-view-links

### PR DESCRIPTION
**Pull request by liyuanmontreal**
feature:
display link list  , every item with a delete button
when user click delete, emit delete event to parent component

### CODE REVIEW

File: src/components/controls/link/km-view-links.vue [hunk 1]
```diff
@@ -0,0 +1,28 @@
+<template>
+    <table class="table table-striped" v-if="links?.length"> 
+        <tbody>
+            <tr  v-for="(item, index) in links" :key="index">  
+                <td>{{languages[item.language]}}  <a :href="item.url" >{{item.name}}</a></td>
+                <td class = "text-end"><span class="btn" @click="deleteLink(index)"><i class="fa fa-trash-o icon"></i></span></td>
```
Comment (blaisefonseca) [inline src/components/controls/link/km-view-links.vue]: please use bootstrap icons

**Review by blaisefonseca — APPROVED** [review]

File: src/components/controls/link/km-view-links.vue [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        </tbody>
+    </table> 
+</template>
+
+<script setup>
+  import { defineEmits, computed } from 'vue'
+  import { languages } from '../../../data/language'
+
+  const model = defineModel({type:Array, required:true});
+  const emit = defineEmits(['deleteLink'],['editLink']);
```
Comment (blaisefonseca) [inline src/components/controls/link/km-view-links.vue]:
```suggestion
  const emit = defineEmits(['deleteLink', 'editLink']);
```

File: src/components/controls/link/km-view-links.vue (hunk 1 above)
Comment (liyuanmontreal) [reply src/components/controls/link/km-view-links.vue]: done

File: src/components/controls/link/km-view-links.vue (hunk 2 above)
Comment (liyuanmontreal) [reply src/components/controls/link/km-view-links.vue]: done

File: src/components/controls/preview.vue [hunk 3]
```diff
@@ -26,13 +26,36 @@
...
+        <div class="row">
+            <div class="col-12">
+                <div class="card">
+                    <div class="card-header">
+                        Km view links
+                    </div>
+                    <div class="card-body"> 
+                        <div class="row">
+                            <div class="col-6"> 
+                                <km-view-links v-model="links"   @onEdit = "edit($event)"  @onDelete = "remove($event)"></km-view-links>
```
Comment (blaisefonseca) [inline src/components/controls/preview.vue]:
```suggestion
                                <km-view-links v-model="links"   @on-edit = "edit($event)"  @on-delete = "remove($event)"></km-view-links>                                
```

File: src/components/controls/preview.vue [hunk 4]
```diff
@@ -26,13 +26,36 @@
...
+                    </div>
+                    <div class="card-body"> 
+                        <div class="row">
+                            <div class="col-6"> 
+                                <km-view-links v-model="links"   @onEdit = "edit($event)"  @onDelete = "remove($event)"></km-view-links>                                
+                            </div>
+                            <div class="col-6">
+                                <div class="callout callout-warning">                              
+                                    <code id="kmViewLinks">                             
+                                        &lt;km-view-links v-model=&quot;links&quot;    @onEdit  = &quot;edit($event)&quot;  @onDelete = &quot;remove($event)&quot;&gt;&lt;/km-view-links&gt;
```
Comment (blaisefonseca) [inline src/components/controls/preview.vue]:
```suggestion
                                        &lt;km-view-links v-model=&quot;links&quot;    @on-edit  = &quot;edit($event)&quot;  @on-delete = &quot;remove($event)&quot;&gt;&lt;/km-view-links&gt;                         
```

**Pull request merged**